plugins/commands/snapshot/command/save.rb

```diff
@@ -28,7 +33,22 @@ def execute
 
           name = argv.pop
           with_target_vms(argv) do |vm|
-            vm.action(:snapshot_save, snapshot_name: name)
+            if !vm.provider.capability?(:snapshot_list)
+              vm.ui.info(I18n.t("vagrant.commands.snapshot.not_supported"))
+              next
+            end
+
+            snapshot_list = vm.provider.capability(:snapshot_list)
+
+            if !snapshot_list.include? name
+              vm.action(:snapshot_save, snapshot_name: name)
+            elsif options[:force]
+              # not a unique snapshot name
+              vm.action(:snapshot_delete, snapshot_name: name)
+              vm.action(:snapshot_save, snapshot_name: name)
+            else
+              vm.ui.warn(I18n.t("vagrant.commands.snapshot.force"))
```

Comment:
This should raise an exception, forcing a non-zero exit code providing feedback that the snapshot was not taken, rather than issuing a warning.